### PR TITLE
[FIX] GCC8 Warning

### LINKED ISSUE #2353: GCC 8 warning in nightly builds
See [CDash](http://cdash.seqan.de/viewBuildError.php?type=1&buildid=289832)

```
/.../Darwin-16.7.0_g++-mp-8_32/include/seqan/basic/iterator_adaptor.h:102:9: warning: '<anonymous>' is used uninitialized in this function [-Wuninitialized]

         data_iterator = TIterator();
```

### PR DESCRIPTION
**Pull request by eseiler**
Resolves #2353

### CODE REVIEW

File: include/seqan/basic/iterator_adaptor.h [hunk 1]
```diff
@@ -96,10 +96,10 @@ class Iter<TContainer, AdaptorIterator<TIterator, TSpec> >
     // Constructors
     // ------------------------------------------------------------------------
 
-   
-    Iter() : data_container()
+
+    Iter() : data_container(),
+             data_iterator()
     {
```
Comment (smehringer) [inline include/seqan/basic/iterator_adaptor.h]: put the closing bracket on the same line if content is empty

Comment (eseiler) [reply include/seqan/basic/iterator_adaptor.h]: The other constructors also put the closing bracket on the next line 🤔

Comment (smehringer) [reply include/seqan/basic/iterator_adaptor.h]: True, but this does not follow the seqan [style guides](https://seqan.readthedocs.io/en/master/Infrastructure/Contribute/StyleCpp.html#constructor-initialization-lists) so when you touch the code I think it's better to do it "the right way"?

**Comment by smehringer** [comment]
!rebuild

**Pull request merged**